mailet/icalendar/src/test/java/org/apache/james/transport/mailets/model/ICALTest.java

```diff
@@ -0,0 +1,130 @@
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.transport.mailets.model;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.apache.mailet.MailAddress;
+import org.apache.mailet.base.MailAddressFixture;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import net.fortuna.ical4j.data.CalendarBuilder;
+import net.fortuna.ical4j.model.Calendar;
+
+public class ICALTest {
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @Test
+    public void buildShouldFailWhenNoCalendar() throws Exception {
+        expectedException.expect(NullPointerException.class);
+
+        ICAL.builder()
+            .recipient(MailAddressFixture.ANY_AT_JAMES)
+            .sender(MailAddressFixture.OTHER_AT_JAMES)
+            .build();
+    }
+
+    @Test
+    public void buildShouldFailWhenNoSender() throws Exception {
+        expectedException.expect(NullPointerException.class);
+
+        Calendar calendar = new CalendarBuilder().build(ClassLoader.getSystemResourceAsStream("ics/meeting.ics"));
+
+        ICAL.builder()
+            .recipient(MailAddressFixture.ANY_AT_JAMES)
+            .from(calendar)
+            .build();
+    }
+
+    @Test
+    public void buildShouldFailWhenNoRecipient() throws Exception {
+        expectedException.expect(NullPointerException.class);
+
+        Calendar calendar = new CalendarBuilder().build(ClassLoader.getSystemResourceAsStream("ics/meeting.ics"));
+
+        ICAL.builder()
+            .sender(MailAddressFixture.OTHER_AT_JAMES)
+            .from(calendar)
+            .build();
+    }
+
+
+    @Test
+    public void buildShouldWork() throws Exception {
+        Calendar calendar = new CalendarBuilder().build(ClassLoader.getSystemResourceAsStream("ics/meeting.ics"));
+
+        MailAddress recipient = MailAddressFixture.ANY_AT_JAMES;
+        MailAddress sender = MailAddressFixture.OTHER_AT_JAMES;
+        ICAL ical = ICAL.builder()
+            .recipient(recipient)
+            .sender(sender)
+            .from(calendar)
+            .build();
+
+        assertThat(ical.getRecipient()).isEqualTo(recipient.asString());
+        assertThat(ical.getSender()).isEqualTo(sender.asString());
+        assertThat(ical.getUid())
+            .contains("f1514f44bf39311568d640727cff54e819573448d09d2e5677987ff29caa01a9e047feb2aab16e43439a608f28671ab7" +
+                "c10e754ce92be513f8e04ae9ff15e65a9819cf285a6962bc");
+        assertThat(ical.getMethod()).contains("REQUEST");
+        assertThat(ical.getRecurrenceId()).isEmpty();
+        assertThat(ical.getDtstamp()).contains("20170106T115036Z");
+        assertThat(ical.getSequence()).contains("0");
+        assertThat(ical.getIcal()).isEqualTo("BEGIN:VCALENDAR\r\n" +
```

Comment:
what is the point of asserting the whole ical ? you could check against the same file or assert against a hash